internal/validators/namespace.go

```diff
@@ -107,6 +118,46 @@ func (v *Namespace) handle(req *nsRequest) admission.Response {
 	return allow("")
 }
 
+// illegalTreeLabel checks if tree labels are being created or modified
+// by any user or service account since only HNC service account is
+// allowed to do so
+func (v *Namespace) illegalTreeLabel(req *nsRequest) admission.Response {
+	msg := "Only HNC Service account can modify tree labels"
+
+	if req.op == k8sadm.Delete {
+		return allow("")
+	}
+
+	oldLabels := map[string]string{}
+	if req.oldns != nil {
+		oldLabels = req.oldns.Labels
+	}
+
+	for key, val := range req.ns.Labels {
+		if oldVal, ok := oldLabels[key]; !ok {
+			if strings.Contains(key, api.LabelTreeDepthSuffix) {
+				return deny(metav1.StatusReasonForbidden, msg)
```

Comment:
It would be nice to say something like:

`"Cannot set or modify tree label %q in namespace %q; these can only be modified by HNC."`

E.g. if this is being called as a result of `kubectl apply -f <directory>`, then the user might not know _which_ namespace in that directory is the problem. And without printing the tree label itself, the user might not know what label we're talking about if they're not HNC experts.

I'd rather not say "Only the HNC service account can do this" because that implies that their mistake was that they're not running as the right service account. But the problem is that they shouldn't be attempting to set these labels at all.